libraries/Microsoft.Bot.Schema/Teams/Generated/MessagingExtensionAction.cs

```diff
@@ -44,14 +44,16 @@ public MessagingExtensionAction()
         /// <param name="botActivityPreview">A collection of bot activities.</param>
         /// <param name="messagePayload">Message content sent as part of the
         /// command request.</param>
-        public MessagingExtensionAction(object data = default(object), TaskModuleRequestContext context = default(TaskModuleRequestContext), string commandId = default(string), string commandContext = default(string), string botMessagePreviewAction = default(string), IList<Activity> botActivityPreview = default(IList<Activity>), MessageActionsPayload messagePayload = default(MessageActionsPayload))
+        /// <param name="state">State parameter passed back to the bot after authentication flow.</param>
+        public MessagingExtensionAction(object data = default(object), TaskModuleRequestContext context = default(TaskModuleRequestContext), string commandId = default(string), string commandContext = default(string), string botMessagePreviewAction = default(string), IList<Activity> botActivityPreview = default(IList<Activity>), MessageActionsPayload messagePayload = default(MessageActionsPayload), string state = default(string))
```

Comment:
This number of parameters is not recommended, and as we need more properties, the list will grow. We should introduce a method that receives an object containing all the data needed. If backward compat is an issue, maybe an overload?

Reply:
Agreed!  I think the plan is to create factory methods for this, and other Teams specific invoke responses.

That said, we do have a bunch of ridiculous and unruly constructors throughout the codebase: https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Schema/Activity.cs#L125

Reply:
Removed the new constructor parameter.

It is a shame we even have these constructors.